lib/dns.js

```diff
@@ -239,11 +242,19 @@ function onresolve(err, result, ttls) {
     this.callback(null, result);
 }
 
+// Resolver instances correspond 1:1 to c-ares channels.
+class Resolver {
+  constructor() {
+    this._handle = new ChannelWrap();
```

Comment:
Using a plain `_handle` property is kind of the standard in Node core. If we were to hide something, I’d suggest thinking of a meaningful custom `util.inspect` overload for resolvers.